File: problems/problem_007.py
"""
By listing the first six prime numbers: 
2, 3, 5, 7, 11, and 13,
we can see that the 6th prime is 13.

What is the 10001st prime number?

"""


def prime(x):
    for n in range(2, int(x / 2) + 1):
        if (x % n) == 0:
            return False
    else:
        return True


def answer():
    primes = []
    num = 2

    while len(primes) < 10001 :
        if prime(num):
            primes.append(num)
        num += 1
    return primes[-1]
    
    
print(answer())
    
    ### Answer is 104743 ###

File: problems/Javascript Solutions/Problem_007.js
/**
By listing the first six prime numbers: 
2, 3, 5, 7, 11, and 13,
we can see that the 6th prime is 13.

What is the 10001st prime number?
 */

function prime(x) {
    for (let n = 2; n < x; n++ ) {
        if (x % n === 0) {
            return false;
        }
    }
    return true;
}

const stNumber = 10001;

function answer(stNumber) {
    let primes = [];
    let num = 2;

    while (primes.length < stNumber) {
        if (prime(num)) {
            primes.push(num);
        }
        num += 1;
    }
    return primes.slice(-1).pop()
}

console.log(answer(stNumber));

    /* Answer is 104743 */